pkg/controller/storagecluster/storagecluster.go

```diff
@@ -405,6 +406,56 @@ func (c *Controller) validateCloudStorageLabelKey(cluster *corev1.StorageCluster
 	return nil
 }
 
+func (c *Controller) runPreflightCheck(cluster *corev1.StorageCluster) error {
+	check, ok := cluster.Annotations[pxutil.AnnotationPreflightCheck]
+	if !ok || check == "skip" {
+		return nil
+	} else if check == "false" {
+		if cluster.Status.Phase == string(corev1.ClusterConditionTypePreflight)+string(corev1.ClusterOperationFailed) {
+			return fmt.Errorf("preflight check failed, please run it again")
+		}
+		return nil
+	}
+
+	// Only do the preflight check on demand once a time
+	toUpdate := cluster.DeepCopy()
+	toUpdate.Annotations[pxutil.AnnotationPreflightCheck] = "false"
+	var err error
+
+	// Do the preflight check for eks only for now to check the cloud drive permission
+	if preflight.IsEKS() {
+		if err = preflight.Instance().CheckCloudDrivePermission(); err != nil {
+			logrus.WithError(err).Errorf("permission check for eks cloud drive failed")
+		} else {
+			logrus.Infof("JLIAO: permission check for eks cloud drive passed")
```

Comment:
Remove or convert to a "proper" log line

Reply:
don't worry about all log lines, will do before merge.